### PR TITLE
Remove temporary hacks from new NuGet

### PR DESCRIPTION
**Pull request by piotrpMSFT**
@eerhardt @brthor

### CODE REVIEW

File: build/Microsoft.DotNet.Cli.Compile.targets [hunk 1]
```diff
@@ -70,9 +70,6 @@
         <BinObj Include="$(SrcDirectory)/**/bin/**/*" />
         <BinObj Include="$(SrcDirectory)/**/obj/**/*" />
 
-        <FilesToRemoveFromStageDirectory Remove="*" />
-        <FilesToRemoveFromStageDirectory Include="$(StageDirectory)/**/*" />
```
Comment (eerhardt) [inline build/Microsoft.DotNet.Cli.Compile.targets]: I think this was in there before the new NuGet build came in.

File: build/compile/Microsoft.DotNet.Cli.LzmaArchive.targets [hunk 2]
```diff
@@ -23,11 +23,6 @@
 
       <Message Text="Publishing Archiver" />
 
-      <!-- TEMP HACK: need to restore with the current stage, until we get a Stage0 with the new NuGet build -->
```
Comment (brthor) [inline build/compile/Microsoft.DotNet.Cli.LzmaArchive.targets]: Do we need to replace this with restore from a stage0 dotnet?

File: build/Microsoft.DotNet.Cli.Compile.targets (hunk 1 above)
Comment (piotrpMSFT) [inline build/Microsoft.DotNet.Cli.Compile.targets]: Correct. It's generating unnecessary spew in the msbuild logs.

File: build/compile/Microsoft.DotNet.Cli.LzmaArchive.targets (hunk 2 above)
Comment (piotrpMSFT) [inline build/compile/Microsoft.DotNet.Cli.LzmaArchive.targets]: We are not deleting the lock files anymore so a re-restore is not required.

Comment (eerhardt) [inline build/compile/Microsoft.DotNet.Cli.LzmaArchive.targets]: Correct - we only restore once before compile.

File: build/Microsoft.DotNet.Cli.Compile.targets (hunk 1 above)
Comment (eerhardt) [inline build/Microsoft.DotNet.Cli.Compile.targets]: But isn't it necessary to ensure the stage is clean before we start putting stuff in it?

Maybe calling "RemoveDir" instead will cut down on the spew.

**Comment by brthor** [comment]
LGTM :shipit:

File: build/Microsoft.DotNet.Cli.Compile.targets (hunk 1 above)
Comment (eerhardt) [inline build/Microsoft.DotNet.Cli.Compile.targets]: Ah, I see it now on line 83.

**Comment by eerhardt** [comment]
:shipit:

**Pull request merged**